include/swift/AST/Types.h

```diff
@@ -2337,12 +2329,14 @@ class AnyFunctionType : public TypeBase {
     ParameterTypeFlags Flags = {};
     
   public:
-    Type getType() const { return Ty; }
+    Type getType() const;
     CanType getCanType() const {
-      assert(Ty->isCanonical());
-      return CanType(Ty);
+      assert(getType()->isCanonical());
+      return CanType(getType());
     }
     
+    Type getRawType() const { return Ty; }
```

Comment:
This function will go away anyhow.  It's mostly useful in copy-esque constructor calls.  I'll go with `getPlainType` in the mean time.